schema_bump_dry_run_scripts/ontologies/ontology_bump_dry_run.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+
+import gzip
+import json
+import os
+import requests
+
+API_URL = {
+    "prod": "https://api.cellxgene.cziscience.com",
+    "staging": "https://api.cellxgene.staging.single-cell.czi.technology",
+    "dev": "https://api.cellxgene.dev.single-cell.czi.technology",
+}
+
+
+def dry_run():
+    # Load processed ontologies file
+    ontologies = "cellxgene_schema_cli/cellxgene_schema/ontology_files/all_ontology.json.gz"
+    with gzip.open(ontologies, "r") as f:
+        onto_map = json.loads(f.read().decode("utf-8"))
+
+    # fetch all currently published datasets
+    # TODO: include datasets from private collections (requires curator auth, do not upload to GHA)
+    base_url = API_URL[os.getenv("corpus_env", default="dev")]
+    datasets = requests.get(f"{base_url}/curation/v1/datasets").json()
+    # dataset metadata fields that contain ontology terms
+    ontology_types = {
+        "assay",
+        "cell_type",
+        "development_stage",
+        "disease",
+        "organism",
+        "self_reported_ethnicity",
+        "sex",
+        "tissue",
+    }
+    # cache terms we know are not deprecated to skip processing; init with special-case, non-ontology terms we use
+    non_deprecated_term_set = {"multiethnic", "unknown", "na"}
+    f = open("ontologies-curator-report.txt", "w")
```

Comment:
is there a reason you aren't using a context manager for this open file? Best practice is to us a context manager in part because it will close the file even if an except occurs within the context block.